### DISCUSSION TITLE
[Nextcloud] Are contacts not integrated automatically?

### DISCUSSION THREAD

**Discussion by unalignedcoder** [the-djmaze/snappymail#923] [General]
I am just trying this through the Nextcloud app. 

Honestly the only attraction in trying this is the promise that Nextcloud contacts are integrated.

From the app description:
`Integration with other Nextcloud apps (Contacts, Files and Calendar)`

However, I cannot see this function reflected anywhere. 

I have thousands of contacts saved in Nextcloud, but they are not available when composing a message in SnappyMail; similarly, opening a contact from the contacts app does not provide any ability to compose directly a message in SnappyMail.

So, how is this integration exactly achieved?

I am aware that on an admin-basis one can retrieve contacts from a WebDav addressbook, however this is true of any webdav server, not a special integration with Nextcloud. Rainloop already had this functionality.

My expectation would be for this integration to happen automatically, since the user is already logged in into Nextcloud when using the SnappyMail app, and the contacts are but a tab away.

**Comment by cm-schl** [comment]
Hi,
if you try to write a mail and you begin to write inside the "To:" field, your contacts saved in Nextcloud Contacts should be shown after some letters.
If this does not work you should check if in the Admin Panel under "Extensions" the Nextcloud plugin is activated AND if in the configuration of this plugin (cogwheel) the "Suggestions" are activated... 

Starting from Nextcloud Contacts should work because you can tell your browser to use SnappyMail when clicking on a mailto: link. Afterwards simply click on the icon beside the email address in Nextcloud Contacts. SnappyMail should open. I say "should" as I made a test and it did not work well (SnappyMail opened, but no new mail dialog opened). But the system I use for tests is not the best configured as I'm testing a lot functions in SnappyMail).

**Reply by unalignedcoder** [reply]
Thank you, I enabled suggestions and now it seems to be working. It takes it a second to pull some suggestions, but that's understandable if many contacts are present.

However, **it seems to be only pulling suggestions from the main "Contacts" Addressbook?** 

Any way to specify which address books should be looked into?

*

As far as making SnappyMail the default email client, it's not ideal - would rather have a contained integration within the contacts app, without messing with the system - **considering you may be accessing your Nextcloud web interface from any given device.**

**Reply by cm-schl** [reply]
Perfect!
I've read something about a change in a prior version to exclude system addressbooks (inside the config of the plugin). To be honest i read about, but never tried it. Maybe in your configuration this flag is also unchecked and therefore only the personal address book is used? Would be interesting if this is the case.

So your problem could be that you would like that for example Firefox normally opens one program when clicking on a mailto link and you specifically can use SnappyMail out of the contacts app without mailto integration? If true - this is not possible at the moment.
The integration is WIP and you can add requests here https://github.com/the-djmaze/snappymail/issues/569. When you are a programmer surly everyone would be happy if you could contribute something to the plugin as at the moment the most work is done by @the-djmaze .

**Reply by unalignedcoder** [reply]
Yes, I had expected SnappyMail to simply make an appearance in the Contacts app interface, for example by showing an internal link to "Write a message to this contact in SnappyMail". I guess the Contacts app people should be involved? I am not familiar with the Nextcloud internal customs, so I wouldn't know if this is common practice or something hardly ever done.

**Reply by the-djmaze** [reply]
https://support.mozilla.org/en-US/kb/change-program-used-open-email-links

Keep in mind that it does not work in private windows.

**Reply by unalignedcoder** [reply]
> https://support.mozilla.org/en-US/kb/change-program-used-open-email-links
> 
> Keep in mind that it does not work in private windows.

Like I was saying, this solution is not ideal given the portability of the Nextcoud web interface. The user could be accessing it from any number of devices, including devices of which they have no control, so a change to the system or browser settings should not be required.

**Reply by the-djmaze** [reply]
> so a change to the system or browser settings should not be required.

Then you must add a feature request to the Nextcloud Contacts github at https://github.com/nextcloud/contacts
